### ISSUE TITLE
Timezone not working

### ISSUE THREAD

**Issue by rafante** [stablekernel/aqueduct#871]
Hi! 
I'm having a problem that I can't figure out a way to solve.

I have some models with a DateTime property and when building the ManagedContext, I'm using timeZone: "America/Sao_Paulo" on PostgreSQLPersistentStore.fromConnectionInfo

```dart
final psc = PostgreSQLPersistentStore.fromConnectionInfo(
      config.database.username,
      config.database.password,
      config.database.host,
      config.database.port,
      config.database.databaseName,
      timeZone: 'America/Sao_Paulo',
    );

final dataModel = ManagedDataModel.fromCurrentMirrorSystem();
return ManagedContext(dataModel, psc);
```

In addition, I'm setting the database parameter TimeZone to America/Sao_Paulo:

![Screen Shot 2020-07-04 at 13 20 12](https://user-images.githubusercontent.com/4541914/86516673-1a103e80-bdf9-11ea-8a6f-dc26b4ddb722.png)

Turns out that whenever I try to save a DateTime it stores it on the database with a difference of 3 hours and I have to calculate the difference, but there's always a bug here or there and my customer is no doubting of the confidence of the reports, could you please help me out?

Thank you, so much!

Good bye

**Comment by Reductions** [comment]
Aqueduct always writes the time in UTC. You can transform the date time in whatever timezone you want in the backend using DateTime methods.

If you can give me a better example of how are you using the timestamps I might be able to give you a better advice on how and where to make the transformation.

**Comment by rafante** [comment]
> Aqueduct always writes the time in UTC. You can transform the date time in whatever timezone you want in the backend using DateTime methods.
> 
> If you can give me a better example of how are you using the timestamps I might be able to give you a better advice on how and where to make the transformation.

Right, I have for example a controller for the model Agendamento (scheduling in portuguese) and on the put method the user can re-schedule the Agendamento, say for example for 2 hours later, like so:

```dart
@Operation.put('id')
  Future<Response> alterar(
      @Bind.path('id') int id, @Bind.body() Agendamento objeto) async {
    //Preenche o coletador default, se nenhum for informado
    if (objeto.coletador == null) {
      final coletadorDefaultQuery = Query<Coletador>(context)
        ..where((c) => c.isDefault).equalTo(true);
      final Coletador coletador = await coletadorDefaultQuery.fetchOne();
      objeto.coletador = coletador;
    }
    final query = Query<Agendamento>(context)
      ..where((obj) => obj.id).equalTo(id)
      ..values = objeto;
    return Response.ok(await query.updateOne());
  }
```

The model is down below:

```dart
class Agendamento extends ManagedObject<_Agendamento> implements _Agendamento {}

class _Agendamento {
  @primaryKey
  int id;

  String descricao;
  int duracaoIntervalos = 1;
  AgendamentoStatus status = AgendamentoStatus.criado;

  @Column(indexed: true)
  DateTime data;

  @Column(nullable: true)
  DateTime ultimaModificacao;

  @Column(nullable: true)
  int usuarioUltimaAlteracao;

  @Column(nullable: true)
  String justificativa;

  @Relate(#agendamentos, isRequired: true, onDelete: DeleteRule.cascade)
  Paciente paciente;

  @Relate(#agendamentos, isRequired: false)
  Usuario usuarioCriador;

  @Relate(#agendamentos)
  Coletador coletador;
}

enum AgendamentoStatus { criado, reagendado, concluido, cancelado, confirmado }
```

The property "data" (date in portuguese) is being filled (on debug mode) correctly. In the example below with "2020-07-07 17:40:00", like down below:

![Screen Shot 2020-07-04 at 14 33 00](https://user-images.githubusercontent.com/4541914/86517923-45982680-be03-11ea-8c88-6d47c049ccad.png)

But in database it saves like this

![Uploading Screen Shot 2020-07-04 at 14.34.21.png…]()

When reading it, it comes with the wrong date saved on the database

![Screen Shot 2020-07-04 at 14 36 29](https://user-images.githubusercontent.com/4541914/86517983-c2c39b80-be03-11ea-9144-5bfb0115bc8a.png)

I need some help because I have datetimes in many models and There's always one with the problem and I don't see, but the customer does.

**Comment by Reductions** [comment]
So the data you read from the database is "2020-07-07 20:40:00.00Z" where the "Z" signifies that the DateTime is in UTC.

I don't know how do you present it to the user but if you have a frontend it is their job to transform from UTC to local time. If you as local time for the user might be different thing.

If you know that the sever time is the same as the one the client wants to use you could do: date.toLocal() -> this converts the date to local time of the server.

**Comment by Reductions** [comment]
So one more question are you creating and API product or a backend for your own backend?
